zubhub_frontend/zubhub/src/components/notification/Notification.jsx

```diff
@@ -0,0 +1,140 @@
+    let plural = '';
+    let timeType = '';
+    var type;
+
+    if (t < minute) {
+      // seconds
+      timeType += 'second';
+      type = second;
+    } else if (t >= minute && t < hour) {
+      // minutes
+      timeType += 'minute';
+      type = minute;
+    } else if (t >= hour && t < day) {
+      // hours
+      timeType += 'hour';
+      type = hour;
+    } else if (t >= day && t < week) {
+      // days
+      timeType += 'day';
+      type = day;
+    } else if (t >= week && t < month) {
+      // weeks
+      timeType += 'week';
+      type = week;
+    } else if (t >= month && t < year) {
+      // months
+      timeType += 'month';
+      type = month;
+    } else if (t > year) {
+      // render years ago
+      timeType += 'year';
+      type = year;
+    }
+    const num = Math.max(1, Math.round(t / type));
+    if (num > 1) plural += 's';
+    return num.toString() + ' ' + timeType + plural + ' ago';
+  };
+
+  // method for rendering message
+  const getNotification = () => {
+    if (notification.type == NOTIFICATION_TYPE.CLAP) {
+      return (
+        <>
+          <strong>{notification.source.username}</strong> clapped for "
+          {notification.projectName}"
+        </>
+      );
+    } else if (notification.type == NOTIFICATION_TYPE.COMMENT) {
+      return (
+        <>
+          <strong>{notification.source.username}</strong> commented on "
+          {notification.projectName}!"
+        </>
+      );
+    } else if (notification.type == NOTIFICATION_TYPE.BOOKMARK) {
+      return (
+        <>
+          <strong>{notification.source.username}</strong> bookmarked "
+          {notification.projectName}!"
+        </>
+      );
+    } else if (notification.type == NOTIFICATION_TYPE.FOLLOW) {
+      return (
+        <>
+          <strong>{notification.source.username}</strong> started following you!
+        </>
+      );
+    } else if (notification.type == NOTIFICATION_TYPE.FOLLOWING_PROJECT) {
+      return (
+        <>
+          <strong>{notification.source.username}</strong> posted a new project!
+        </>
+      );
+    }
+  };
+
+  return (
+    <>
+      <div
+        className={classes.notificationStyle}
+        onClick={() => {
+          if (!notification.viewed) {
+            viewNotification({
+              id: notification.id,
+              token: token,
+            });
+          }
+          window.location.href = notification.link;
+        }}
+      >
+        <img
+          className={classes.image}
+          src={notification.source.avatar}
+          alt="user-profile"
+        />
+
+        <div className={classes.text}>
+          <p className={classes.message}>{getNotification()}</p>
+          <p className={classes.time}>{renderTimeAgo()}</p>
+        </div>
+        {!notification.viewed && <div className={classes.viewDot}></div>}
+        {notification.viewed && <div className={classes.unviewed}></div>}
+      </div>
```

Comment:
Great work on the css! However, we want to try to use as many Material UI components as possible so that we don't have to do a lot of the styling ourselves. I quickly wrote up the following block of code

```
<ListItem
        alignItems="flex-start"
        disableGutters
        button
        className={classes.notificationStyle}
      >
        <ListItemAvatar>
          <Avatar
            src={notification.source.avatar}
          />
        </ListItemAvatar>
        <ListItemText
          primary={getNotification()}
          secondary={renderTimeAgo()}
          style={{ color: 'black', width: '275px' }}
        />
        /* notification icon div */
</ListItem>
```

based on this doc (https://v4.mui.com/components/lists/#folder-list) and you'll notice that it accomplishes a core part of the UI with very minimal styling of our own. If you use this example and some of our previous styling you should be able to achieve the same styling in a cleaner fashion.